COMPILING.md

```diff
@@ -1,19 +1,125 @@
 # WHAT ARCHITECTURE?
 
-CPROVER now needs a C++11 compliant compiler and works in the following
-environments:
+CPROVER now needs a C++11 compliant compiler and is known to work in the
+following environments:
 
 - Linux
 - MacOS X
+- Microsoft Visual Studio 2019
+
+The above environments are currently tested as part of our continuous
+integration system. It separately tests both the CMake build system and the
+hand-written make files. The latest build steps being used in CI can be
+[referenced here](https://github.com/diffblue/cbmc/blob/develop/.github/workflows/pull-request-checks.yaml).
+
+The environments below have been used successfully in the
+past, but are not actively tested:
+
 - Solaris 11
 - FreeBSD 11
 - Cygwin
-- Microsoft Visual Studio
 
-The rest of this document is split up into three parts: compilation on Linux,
-MacOS, Windows.  Please read the section appropriate for your machine.
+# Building using CMAKE
+
+Building with CMake is supported across Linux, MacOS X and Windows with Visual
+Studio 2019. There are also hand-written make files which can be used to build
+separate binaries independently. Usage of the hand-written make files is
```

Comment:
I think this might be clearer as something like "can be used to build (sub) components of the cbmc package" or something like this? I agree the current version is not clear (I don't know if mi version is any better). But the spirit I think is we could build `goto-diff` or `jbmc` without building all the others?

Reply:
But ... that's no different with cmake, is it? You can easily build individual components by just specifying them as targets (e.g., I sometimes do `ninja -C build cbmc` when I don't care about building all the rest).

Reply:
I agree. I'm guessing at what the wording meant to convey.